### ISSUE TITLE
Find an appropriate solution for displaying images

### ISSUE THREAD

**Issue by nicodh** [deltachat/deltachat-desktop#4410]
We need a solution to meet following requirements:

- [ ] the height of an image must be set even before the image is loaded (core provides dimensions in the message object)
- [ ] image-only messages should not show (extreme) padding except for very narrow images
- [ ] define a range to avoid too big and too small images
- [ ] narrow image-only messages should display a footer wide enough for timestamp and reactions

Related to:
- #4404
- PRs #4407 & #4315

**Comment by Simon-Laux** [comment]
my messages on this topic from the group discussion:

maybe we could provide a placeholder empty image in the correct dimensions

so we would let the browser make the layout calculations 

This might work (seems like you don't even need base64 encoding):
```
<img src='data:image/svg+xml;utf8,<svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" version="1.1" x="0px" y="0px" viewBox="0 0 100 100" height="100px" width="200px"></svg>' alt="" />
```
though I'm not sure whats the best way to load the actual image and replace it, maybe an xhr request do that it is cached, then replace the url or sth. or have two elements and remove the placeholder on load

**Comment by nicodh** [comment]
But we have the dimensions except for images "the core could not understand" and videos.

**Comment by nicodh** [comment]
I would suggest: go with the dimensions coming from core and do the needed adaptions in the template (like setting limits to not have too big images or paddings)

**Comment by Simon-Laux** [comment]
yes, but that's an edge case for which we could have a fallback.

And maybe we could tell core to save the dimensions from the ui in these cases? But I think only fallback is just fine.

**Comment by nicodh** [comment]
Let's continue this discussion once we have a concrete proposal

**Comment by Simon-Laux** [comment]
I'm not interested in a discussion here at the time, my goal was to provide more information in the issue thread.

**Comment by nicodh** [comment]
Ok sure - thanks for that!

**Comment by nicodh** [comment]
Possible solution UI wise:

Landscape and portrait ratio images have a fixed width, portrait ratio additionally a max height (cropped if too high)

![image](https://github.com/user-attachments/assets/09f90f27-c690-4de0-9a91-6c1f97ec065f)

Extreme portrait images are cropped to a fix minimal width and maximal height:
Result:
![image](https://github.com/user-attachments/assets/721f43d9-076e-4e20-a787-784d4f22a0f0)

Original:
![image](https://github.com/user-attachments/assets/a8faec58-2d3d-45c0-b7fc-c968734b0bc6)

The minimal widht should consider possible reactions and the timestamp in the footer

**Comment by iequidoo** [comment]
> Possible solution UI wise: [...]

Looks like a good solution for me as a Desktop user. But reading it, it seems to me that you even don't need to classify images as landscape and portrait ones implementation-wise, basically you just have min_width, max_width and max_height. But i'd say that min_height makes sense too to avoid very narrow horizontal panoramas (better to crop them too apparently). I'd say that min_height should be `max_height / max_width * min_width` (to have the same ratio limit), but min_height may be a bit less because human eye perceives vertical sizes as slightly larger.

**Comment by nicodh** [comment]
> it seems to me that you even don't need to classify images as landscape and portrait ones implementation-wise

this is was more meant for easier communication - and yes a minimal height for extreme panorama ratios makes sense too

In general the purpose of this comment is to find a consensus how a solution should look like, before implementing it :-)

**Comment by WofWca** [comment]
Looks like we used to have some JS dimensions calculations:

https://github.com/deltachat/deltachat-desktop/blob/d1a3f8dbebe4cfa9f74ec4e53e29637eb1e7c573/packages/frontend/src/components/attachment/messageAttachment.tsx#L74-L76

**Comment by Simon-Laux** [comment]
for the case that the images/media do not have the correct dimensions we can overwrite the media size in core with the [`latefiling_mediasize`](https://github.com/deltachat/deltachat-core-rust/blob/5dc8788eab35befa66a292f46d94f093183ca6a5/src/message.rs#L1151) method. Though this method is not yet exposed through the jsonrpc api.
